### PR TITLE
Add the feature to auto-scroll

### PR DESCRIPTION
**Pull request by buszk**
As a music lyric software, it is pretty inconvenient not having the feature to scroll down the lyric box automatically. I recorded the duration into the song object and used the window height and total scroll height as indicators to decide when to begin scrolling and what the interval should be for scroll down one px. 

This approach is not accurate but approximate because I assume the constant pace of the music lyrics. 

Hope you can add this to the settings of the program so that users can choose it.

### CODE REVIEW

**Comment by thesilent33** [comment]
Yes please. Please add scrolling feature. Thanks. :)

File: lyricfier.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (emilioastarita) [inline lyricfier.iml]: please remove this file.

File: render/Searcher.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import {SearchLyrics} from './plugins/SearchLyrics';
 import {SearchWikia} from "./plugins/SearchWikia";
-
```
Comment (emilioastarita) [inline render/Searcher.ts]: not relevant to the PR

File: render/SongRender.ts [hunk 3]
```diff
@@ -88,15 +88,35 @@ export class SongRender {
                       song.sourceName = result.sourceName;
                     }
                     this.displaySong(song);
-                    this['$nextTick'](() => {
-                        document.getElementById("lyricBox").scrollTop = 0;
-                    });
+                    console.log(this.settings.autoScroll);
+                    if(this.settings.autoScroll) {
+                        this['$nextTick'](() => {
+                            document.getElementById("lyricBox").scrollTop = 0;
```
Comment (emilioastarita) [inline render/SongRender.ts]: could you move this logic to a new function? and only do the work if the setting is enabled.

File: render/SongRender.ts [hunk 4]
```diff
@@ -88,15 +88,35 @@ export class SongRender {
...
+                            var scrollHeight = document.getElementById("lyricBox").scrollHeight;
```
Comment (emilioastarita) [inline render/SongRender.ts]: use `const` or `let` to declare local vars.

File: render/SongRender.ts [hunk 5]
```diff
@@ -88,15 +88,35 @@ export class SongRender {
...
+                            setTimeout(this.pageScroll.bind(null, this.song, scrollHeight - height, scrollInterval), waitTime);
+                        });
+                    }
                     this.scheduleNextCall();
                 });
             }
         });
     }
 
+    pageScroll(aSong, i, s) {
```
Comment (emilioastarita) [inline render/SongRender.ts]: please rename this vars with better names.

File: render/SpotifyService.ts [hunk 6]
```diff
@@ -80,12 +80,14 @@ export class SpotifyService {
             return cb(null, this.csrfToken);
         }
         const url = this.url('/simplecsrf/token.json');
+        console.log("========");
```
Comment (emilioastarita) [inline render/SpotifyService.ts]: remove this

**Review by emilioastarita — CHANGES_REQUESTED** [review]
Hello @bruceshenzk thanks for your PR. I need some changes to merge this. Could you please work on my comments? If you need any help please let me know. And sorry for the delay.

**Comment by buszk** [comment]
Hi, @emilioastarita . Please review the changes.

**Review by emilioastarita — APPROVED** [review]

**Pull request merged**